common/src/main/java/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -188,8 +188,12 @@ class DeviceCache(
 
     @Synchronized
     fun getPreviouslySentHashedTokens(): Set<String> {
-        return (preferences.getStringSet(tokensCacheKey, emptySet())?.toSet() ?: emptySet()).also {
-            log(LogIntent.DEBUG, ReceiptStrings.TOKENS_ALREADY_POSTED.format(it))
+        return try {
+            (preferences.getStringSet(tokensCacheKey, emptySet())?.toSet() ?: emptySet()).also {
+                log(LogIntent.DEBUG, ReceiptStrings.TOKENS_ALREADY_POSTED.format(it))
+            }
+        } catch (e: Exception) {
```

Comment:
Thanks so much for opening this! It makes sense. Can we catch `ClassCastException` instead? Exception is too generic and we could be missing other exceptions.

Reply:
Yep!